sys/mips/mips/pm_machdep.c

```diff
@@ -722,6 +722,7 @@ exec_setregs(struct thread *td, struct image_params *imgp, uintcap_t stack)
 
 		td->td_frame->t9 = imgp->entry_addr & ~3; /* abicall req */
 #if __has_feature(capabilities)
+		td->td_md.md_tls = NULL;
```

Comment:
It needs to be cleared for !CHERI too.  I think it would be cleanest to set it to NULL lower in the file right above the code that sets md_tls_tcb_offset (line 768 or so).  fork() has the two lines that copy the two fields together already.

Reply:
The handling of md_tls_tcb_offset is confusing. We set it here, but also again in cpu_set_user_tls? I don't see why we bother to do the latter, it should be constant after an exec until the next exec.

Reply:
I do think it should be dropped from cpu_set_user_tls.  That is an upstream bug though.  The lack of clearing md_tls is also an upstream bug it seems.

Reply:
Yes, although we are the source of the cpu_set_user_tls inefficiency (https://github.com/freebsd/freebsd/commit/d2ad1cefd89b5abc351574f7ad3777c049930902) :)

Reply:
I can fix the `cpu_set_user_tls` inefficiency here as well, do we care about upstreaming these?

Reply:
I'm already testing a branch with the changes pending for upstream. :)